src/simulation/LoadGenerator.cpp

```diff
@@ -1662,54 +1669,36 @@ std::pair<SorobanResources, uint32_t>
 LoadGenerator::sorobanRandomUploadResources()
 {
     auto const& cfg = mApp.getConfig();
-    Resource maxPerTx =
-        mApp.getLedgerManager().maxSorobanTransactionResources();
-
     SorobanResources resources{};
-    resources.instructions =
-        sampleDiscrete(cfg.LOADGEN_INSTRUCTIONS_FOR_TESTING,
-                       cfg.LOADGEN_INSTRUCTIONS_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_INSTRUCTIONS);
 
-    // Respect both contract size limit and TX write byte
-    // limits including key overhead
-    uint32_t const keyOverhead = 40;
+    // Sample a random Wasm size
     uint32_t wasmSize = sampleDiscrete(
         cfg.LOADGEN_WASM_BYTES_FOR_TESTING,
         cfg.LOADGEN_WASM_BYTES_DISTRIBUTION_FOR_TESTING, DEFAULT_WASM_BYTES);
-    resources.readBytes =
-        sampleDiscrete(cfg.LOADGEN_IO_KILOBYTES_FOR_TESTING,
-                       cfg.LOADGEN_IO_KILOBYTES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_IO_KILOBYTES) *
-        1024;
-    // Allocate enough write bytes to write the whole Wasm plus the 40 bytes of
-    // the key, plus extra drawn from the distribution.
-    resources.writeBytes =
-        wasmSize + keyOverhead +
-        sampleDiscrete(cfg.LOADGEN_IO_KILOBYTES_FOR_TESTING,
-                       cfg.LOADGEN_IO_KILOBYTES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_IO_KILOBYTES) *
-            1024;
 
-    auto writeKeys = LedgerTestUtils::generateUniqueValidSorobanLedgerEntryKeys(
-        sampleDiscrete(cfg.LOADGEN_NUM_DATA_ENTRIES_FOR_TESTING,
-                       cfg.LOADGEN_NUM_DATA_ENTRIES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_NUM_DATA_ENTRIES));
-
-    for (auto const& key : writeKeys)
-    {
-        resources.footprint.readWrite.emplace_back(key);
-    }
-
-    auto readKeys = LedgerTestUtils::generateUniqueValidSorobanLedgerEntryKeys(
-        sampleDiscrete(cfg.LOADGEN_NUM_DATA_ENTRIES_FOR_TESTING,
-                       cfg.LOADGEN_NUM_DATA_ENTRIES_DISTRIBUTION_FOR_TESTING,
-                       DEFAULT_NUM_DATA_ENTRIES));
+    // Estimate VM instantiation cost, with some additional buffer to increase
+    // the chance that this instruction count is sufficient.
+    uint32_t constexpr vmConstTerm = 451'626;
```

Comment:
Are all these magic numbers something you measured or are some derived from the network config?

Reply:
`vmConstTerm` and `vmLinearTerm` are derived from network config. I changed them to directly use the network config instead in 89c37de4fe9bf25de0770daba3da637d99d08bbd. `vmShiftTerm` is part of the formula for estimating cost that @dmkozh pointed me to. I don't believe it's part of a network config. `vmBufferPerByte` is an additional buffer I added after experimentation to greatly improve the chance that the declared instruction count is sufficient.

I added more comments in 89c37de4fe9bf25de0770daba3da637d99d08bbd to make this more clear.